File: maiworld-coach-web/src/main/webapp/js/service/coachService.js
//服务层

app.service('coachService' ,function($http){
    				this.findAll=function(){
    					return $http.get('../coach/findAll.do');
    				}
    				
    				this.findPage=function(page, rows){
    					return $http.get('../coach/findPage.do?page='+page+'&rows='+rows);
    				}
    				
    				this.findOne=function(cid){
    					return $http.get('../coach/findOne.do?cid='+cid);
    				}
    				
    				this.add=function(entity){
    					return $http.post('../coach/add.do',entity);
    				}
    				
    				this.update=function(entity){
    					return $http.post('../coach/update.do',entity);
    				}
    				
    				this.dele=function(cids){
    					return $http.get('../coach/delete.do?cids='+cids);
    				}
    				
    				this.search=function(page, rows, searchEntity){
    					return $http.post('../coach/search.do?page='+page+'&rows='+rows, searchEntity);
    				}
    			});

File: maiworld-seller-interface/src/main/java/com/maiworld/seller/service/CoachService.java
package com.maiworld.seller.service;

import java.util.List;

import com.maiworld.api.dto.CoachDTO;
import com.maiworld.api.vo.ResultVO;
import com.maiworld.pojo.TbCoach;

import entity.PageResult;

public interface CoachService {

	
	
	public List<TbCoach> findAll();
	
	public PageResult findPage(int pageNum, int pageSize);
	
	public void add(TbCoach coach);
	
	public void update(TbCoach coach);
	
	public TbCoach findOne(Long cid);
	
	public void delete(Long[] cids);
	
	public PageResult findPage( TbCoach coach , int pageNum , int pageSize);
	
	/**
	 * 更新状态
	 * @param cid
	 * @param authentication
	 */
	public void updateAuthentication(Long cid , String authentication);
	
	/**
	 * 查询教练详情
	 * @param openid
	 * @return
	 */
	public ResultVO<CoachDTO> findOne(String openid);
}
